apps/condo/domains/common/components/Table/Filters.tsx

```diff
@@ -1,40 +1,63 @@
+import { GraphQlSearchInput, ISearchInputProps } from '@condo/domains/common/components/GraphQlSearchInput'
+import DatePicker from '@condo/domains/common/components/Pickers/DatePicker'
+import DateRangePicker from '@condo/domains/common/components/Pickers/DateRangePicker'
+import { FilterContainer, SelectFilterContainer } from '@condo/domains/common/components/TableFilter'
 import { colors } from '@condo/domains/common/constants/style'
+import { QueryArgType } from '@condo/domains/common/utils/tables.utils'
+
+import { TrackingEventType, useTracking } from '../TrackingContext'
 
-import { OptionType, QueryArgType } from '../../utils/tables.utils'
-import { GraphQlSearchInput } from '../GraphQlSearchInput'
-import DatePicker from '../Pickers/DatePicker'
-import DateRangePicker from '../Pickers/DateRangePicker'
-import { FilterContainer, SelectFilterContainer } from '../TableFilter'
 
 type FilterIconType = (filtered?: boolean) => React.ReactNode
 type FilterValueType = (path: string | Array<string>, filters: { [x: string]: QueryArgType }) => FilterValue
 
+type CommonFilterDropdownProps = { containerStyles?: CSSProperties }
+type GetCommonFilterDropdownType<P> = (props?: P) => (props: FilterDropdownProps) => React.ReactNode
+
+type GetTextFilterDropdownType = GetCommonFilterDropdownType<{ inputProps?: CustomInputProps } & CommonFilterDropdownProps>
+
+type GetOptionFilterDropdownType = GetCommonFilterDropdownType<{ checkboxGroupProps?: CheckboxGroupProps & { id?: string } } & CommonFilterDropdownProps>
+
+type GetSelectFilterDropdownProps<ValueType> = { selectProps?: CustomSelectProps<ValueType> } & CommonFilterDropdownProps
+type GetSelectFilterDropdownType<ValueType> = GetCommonFilterDropdownType<GetSelectFilterDropdownProps<ValueType>>
+
+type GetGQLSelectFilterDropdownType = GetCommonFilterDropdownType<{ gqlSelectProps?: ISearchInputProps } & CommonFilterDropdownProps>
+
+type GetDateFilterDropdownType = GetCommonFilterDropdownType<{ datePickerProps?: PickerProps<Dayjs> } & CommonFilterDropdownProps>
+
+type GetDateRangeFilterDropdownType = GetCommonFilterDropdownType<{ datePickerProps?: RangePickerProps<Dayjs> } & CommonFilterDropdownProps>
+
+
 const FILTER_DROPDOWN_CHECKBOX_STYLES: CSSProperties = { display: 'flex', flexDirection: 'column' }
 
 export const getFilterIcon: FilterIconType = (filtered) => <FilterFilled style={{ color: filtered ? colors.sberPrimary[5] : undefined }} />
 export const getFilterValue: FilterValueType = (path, filters) => get(filters, path, null)
 
-export const getTextFilterDropdown = (placeholder: string, containerStyles?: CSSProperties) => {
+export const getTextFilterDropdown: GetTextFilterDropdownType = ({ containerStyles, inputProps } = {}) => {
     return ({ setSelectedKeys, selectedKeys, confirm, clearFilters }) => {
         const handleClear = useCallback(() => {
             isFunction(clearFilters) && clearFilters()
-            setSelectedKeys('')
+            setSelectedKeys('' as any)
```

Comment:
please don't use `any`, this method requires argument of type `React.Key[]` which is `type Key = string | number`

Reply:
changed to @ts-ignore and added note about type casting problem